oktapam/client/wrappers/database.go

```diff
@@ -0,0 +1,139 @@
+package wrappers
+
+import (
+	"fmt"
+	"github.com/atko-pam/pam-sdk-go/client/pam"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+)
+
+var (
+	_ ResourceWrapper = (*DatabaseResourceResponseWrapper)(nil)
+	_ ResourceWrapper = (*ManagementConnectionDetailsWrapper)(nil)
+	_ ResourceWrapper = (*MySQLBasicAuthManagementConnectionDetailsWrapper)(nil)
+	_ ResourceWrapper = (*MySQLBasicAuthDetailsWrapper)(nil)
+)
+
+// The key is the single attribute field name, and the value is what will be set for that field.
+type attributeOverrides map[string]any
+
+type ResourceWrapper interface {
+	// AttributeOverridePaths returns the attribute key paths that can be set based on existing values in state.
+	// These are returned recursively, so this should typically only be executed on a top level struct.
+	AttributeOverridePaths() []string
+	// ToResourceMap converts a resource struct into a map of attribute keys to values.
+	// Optional attributeOverrides can be provided to override any resource value.
+	// utils.GenerateAttributeOverrides can be used to generate the required overrides for any resource struct.
+	ToResourceMap(attributeOverrides) map[string]any
+}
+
+type ManagementConnectionDetailsWrapper struct {
+	pam.ManagementConnectionDetails
+}
+type DatabaseResourceResponseWrapper struct {
+	pam.DatabaseResourceResponse
+}
+type MySQLBasicAuthManagementConnectionDetailsWrapper struct {
+	pam.MySQLBasicAuthManagementConnectionDetails
+}
+type MySQLBasicAuthDetailsWrapper struct {
+	pam.MySQLBasicAuthDetails
+}
+
+func (w DatabaseResourceResponseWrapper) ToResourceMap(o attributeOverrides) map[string]any {
+	m := make(map[string]any, 10)
+
+	m[attributes.CanonicalName] = w.CanonicalName
+	m[attributes.DatabaseType] = w.DatabaseType
+	if w.ManagementGatewaySelectorId != "" {
+		m[attributes.ManagementGatewaySelectorID] = w.ManagementGatewaySelectorId
+	}
+	if w.ManagementGatewaySelector != nil {
+		m[attributes.ManagementGatewaySelector] = *w.ManagementGatewaySelector
+	} else {
+		m[attributes.ManagementGatewaySelector] = make(map[string]string)
+	}
+
+	mgmtDetails := make([]any, 1)
+	mgmtDetails[0] = ManagementConnectionDetailsWrapper{w.ManagementConnectionDetails}.ToResourceMap(o)
+
+	m[attributes.ManagementConnectionDetails] = mgmtDetails
+
+	return m
+}
+
+func (w DatabaseResourceResponseWrapper) AttributeOverridePaths() []string {
+	overrides := []string{}
+	childOverrides := ManagementConnectionDetailsWrapper{w.ManagementConnectionDetails}.AttributeOverridePaths()
+	for _, attr := range childOverrides {
+		overrides = append(overrides, fmt.Sprintf("%s.0.%s", attributes.ManagementConnectionDetails, attr))
```

Comment:
same comment for others below, but it would be good to have comments on what each part of the path means.  also, is the '0' value/index guaranteed?

Reply:
done... hopefully it's sufficient.